build-capnp.sh

```diff
@@ -67,7 +67,7 @@ if [[ $1 == "linux" ]]; then
     make -j
     make install-data DESTDIR=capnp-root
 
-elif [[ $1 == "darwin" ]]; then
+elif [[ $1 == "darwin" && $2 == "x64" ]]; then
     # Cross-compile to darwin, with various manual fixes.
 
     CC=o64-clang CXX=o64-clang++ ./configure --build=x86_64-apple-darwin --host=x86_64-linux-gnu
```

Comment:
Hmm. I don't think this is the best approach - it means the entire body of the compilation process is duplicated (see your additions from line 97), including parts which don't depend on the architecture.

I would suggest:

```suggestion
elif [[ $1 == "darwin" ]]; then
    if [[ $2 == "x64" ]]; then
        export CC=o64-clang
        export CXX=o64-clang++
        CONFIGURE_TARGET=x86_64-apple-darwin
    elif [[ $2 == "arm64" ]]; then
        export CC=oa64-clang
        export CXX=oa64-clang++
        CONFIGURE_TARGET=aarch64-apple-darwin
    else
        echo "Invalid ARCH argument; expected 'x64' or 'arm64'"
        exit 1
    fi
    
    # Cross-compile to darwin, with various manual fixes.
    ./configure --build=${CONFIGURE_TARGET} --host=x86_64-linux-gnu
```